src/dialogsettings.cpp

```diff
@@ -376,24 +375,20 @@ void DialogSettings::changeIcon(QToolButton *button)
         QPainter painter(&image);
         svg.render(&painter);
 
-        test = QPixmap::fromImage(image);
+        icon = QPixmap::fromImage(image);
     }
     else
     {
-
-        if ( !test.load( e ) )
-        {
+        if (!icon.load( e ) || Utils::pixmapToString(icon).isEmpty()) {
             QMessageBox::critical(nullptr, tr("Invalid icon"),
                                   tr("Could not load the icon from this file."));
```

Comment:
Well, technically this isn't really a "bug" per se. `libpng` is rejecting to save these images, because (at least for #356) the colorspace contained in the image is invalid with the png standard. From what I read the colorspace was used in the past by some Adobe product but was not standard compliant, which is why `libpng` is reading it correctly but newer version don't want to write it anymore.
But I agree, a separate message would make this more clearer. How about this:
```
Could not load the icon from this file. The image might be stored in an old or unsupported format. Try loading the icon in an image editing tool and saving it in a newer format.
```
I don't know if if that is too technical or not technical enough. What do you think?

Reply:
Could not load the icon from this file because your version of Qt library is unable to process it. You can either upgrade your Qt version to 5.12 or higher, or try to load the icon into an image editing, and save it in different format.

Reply:
The commit I linked above is for Qt 5.**15**.0, which is not released yet so I couldn't test it. It sounded similar so it might fix the problem, but it might also not.

Reply:
> because your version of Qt library is unable to process it.

I'm also not really a fan of that wording, because on Windows the user doesn't have a choice in which Qt version Birdtray is using.

Reply:
```
Could not load the icon from this file. Try loading the icon in an image editing tool and saving it in a different format.
```
Like this?